bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/button/ButtonImpl.java

```diff
@@ -84,6 +84,9 @@
     @Via(type = ResourceSuperType.class)
     private Button button;
 
+    @Self
```

Comment:
For all components `@Self(injectionStrategy = InjectionStrategy.OPTIONAL)` is used but not here? Is that by intention?

Reply:
yes because the ButtonImpl's defaultInjectionStrategy is OPTIONAL.

As I don't have a strong opinion on that but prefer more explicit settings, I will change that, thanks.

Reply:
+1 to change to simply have it consistent